### PR TITLE
Fix handleKindGitHub, Change FindSyncWebhook

### PR DESCRIPTION
**Pull request by susantoscott**
Fix a problem from previous PR where we were returning too early (return when a webhook was found in config). We will always call `FindSyncWebhook` to look for the webhook from GitHub API to determine whether we want to create a new webhook. In the process, we store the webhook in the extsvc `Config`.

TODO: maybe we should store the fact that a repo has a _sync webhook_ so we don't have to call `FindSyncWebhook` every time a job is dequeued (or everytime an external service is [synced](url)). This might be in the form of a boolean attribute on a repo, or storing this repo-webhook pair in the database somewhere.

Also in this PR is changing the `FindSyncWebhook` method to return the complete payload from GitHub instead of just the webhook ID. This is so we can get the `secret` from the `WebhookPayload` `Config` and add it into the extsvc `Config`

## Test plan
`go test -run TestWebhookBuildHandle`

### CODE REVIEW

File: internal/repos/webhook_build_handler.go [hunk 1]
```diff
@@ -62,26 +62,30 @@ func (w *webhookBuildHandler) handleKindGitHub(ctx context.Context, logger log.L
 		return errcode.MakeNonRetryable(errors.Newf("handleKindGitHub: expected *schema.GitHubConnection, got %T", parsed))
 	}
 
+	if webhookExistsInConfig(conn.Webhooks, job.Org) {
+		logger.Info("Webhook found, no need to build new webhook")
```
Comment (indradhanush) [inline internal/repos/webhook_build_handler.go:66]: Maybe add info about the job so that each log line can be distinguished from another.

**Review by indradhanush — COMMENTED** [review]
Looks good to me. Not doing an LGTM as I'd like another pair of eyes on this.

File: internal/repos/webhook_build_handler.go [hunk 2]
```diff
@@ -90,29 +89,24 @@ func (w *webhookBuildHandler) handleKindGitHub(ctx context.Context, logger log.L
...
+	logger.Info("webhook created", log.Int("ID", id))
 	return nil
 }
 
-func webhookExistsInConfig(webhooks []*schema.GitHubWebhook, org string) bool {
-	for _, webhook := range webhooks {
-		if webhook.Org == org {
-			return true
-		}
+func addSecretToExtSvc(svc *types.ExternalService, conn *schema.GitHubConnection, org, secret string) error {
```
Comment (sashaostrikov) [inline internal/repos/webhook_build_handler.go:105]:
```suggestion
func addWebhookToExtSvc(svc *types.ExternalService, conn *schema.GitHubConnection, org, secret string) error {
```

File: internal/repos/webhook_build_handler.go [hunk 3]
```diff
@@ -90,29 +89,24 @@ func (w *webhookBuildHandler) handleKindGitHub(ctx context.Context, logger log.L
...
-
-	id, err = client.CreateSyncWebhook(ctx, job.RepoName, fmt.Sprintf("https://%s", globals.ExternalURL().Host), secret) // TODO: Add to DB
+	id, err := client.CreateSyncWebhook(ctx, job.RepoName, fmt.Sprintf("https://%s", globals.ExternalURL().Host), secret) // TODO: store the webhook
 	if err != nil {
 		return errors.Wrap(err, "handleKindGitHub: CreateSyncWebhook failed")
 	}
 
-	logger.Info(fmt.Sprintf("Created webhook with ID: %d", id))
+	if err := addSecretToExtSvc(svc, conn, job.Org, secret); err != nil {
+		logger.Error("handleKindGitHub: Webhook created but addSecretToExtSvc failed")
```
Comment (sashaostrikov) [inline internal/repos/webhook_build_handler.go:98]: I guess in this case we need a retry. If we log and not return an error, the job will be considered completed. That's why we should return an error in this case (but not the `non-retryable` one)

**Review by sashaostrikov — COMMENTED** [review]
I have a couple of small comments.

Can you also pls add a doc to `handleKindGitHub` function describing step by step what is going on in there -- it will be super helpful

File: internal/repos/webhook_build_handler.go (hunk 3 above)
Comment (susantoscott) [reply internal/repos/webhook_build_handler.go:98]: That makes sense. And I presume the same will be the same for line 79?

Comment (sashaostrikov) [reply internal/repos/webhook_build_handler.go:98]: yep, nice catch!

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Pull request merged**